app/src/main/java/com/example/habitsmasher/HabitEventDialog.java

```diff
@@ -0,0 +1,145 @@
+package com.example.habitsmasher;
+
+import android.app.DatePickerDialog;
+import android.content.Context;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+
+import com.example.habitsmasher.ui.history.HabitEventListFragment;
+
+/**
+ * Abstract UI class that describes any dialog
+ * involving adding and editing Habit Events
+ * @author Jason Kim
+ */
+public abstract class HabitEventDialog extends DialogFragment implements DisplaysErrorMessages {
```

Comment:
The HabitEventDialog class is an abstract class in which EditHabitEventDialog and AddHabitEventDialog extend from

Reply:
So if I'm understanding this right, you've essentially taken all of the shared functionality between the edit and add and made an abstract class, then left out all of the stuff specific to each of them?

Reply:
Yes! Since both classes are basically the same besides a couple of differences, we simply extract the commonalites of the two into a common parent class.